mesonbuild/mtest.py

```diff
@@ -1075,6 +1075,9 @@ async def parse(self, harness: 'TestHarness', lines: T.AsyncIterator[str]) -> No
                 self.additional_error += 'TAP parsing error: ' + i.message
                 res = TestResult.ERROR
 
+        if version is None:
+            self.warnings.append('Unknown TAP version. The first line MUST be `TAP version <int>`. Assuming version 12.')
```

Comment:
That gets repeated for every tests in glib, that clutters a lot the output. But I guess they would need to actually print version, I've got no idea where that's done.

Reply:
I also get a lot of those too:
```
stdout:    5: UNKNOWN: (/home/xclaesse/programmation/glib/builddir/glib/tests/bookmarkfile:240442): GLib-DEBUG: 15:53:49.748: g_set_user_dirs: Setting HOME to /tmp/test_bookmarkfile_HV52W1/bookmarks/load-from-data-dirs/.dirs/home
```

I guess that's every time the unit test prints something to stdout using g_debug(). I think those are the logs @3v1n0 wants to see in https://github.com/mesonbuild/meson/issues/11185.

Reply:
I'm trying to decide whether to print it once because it will usually be the same issue everywhere within a project, or to print it for each case because people would be surprised if they fix the reported issue and then still get the warning somewhere else whack-a-mole style.

Reply:
Those lines should be prefixed with a hash to make them into TAP comments.

Reply:
Made a fix in glib, that could be enough: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3149

Reply:
> Those lines should be prefixed with a hash to make them into TAP comments.

hmm, makes sense I guess. That's probably going to be non trivial to do in glib... But that's not Meson's job indeed.